ledger/eval.go

```diff
@@ -542,6 +542,20 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, ad transacti
 		if dup {
 			return TransactionInLedgerError{txid}
 		}
+
+		// Does the address that authorized the transaction actually match whatever address the sender has rekeyed to?
+		// i.e., the sig/lsig/msig was checked against the txn.Authorizer() address, but does this match the SpendingKey in the sender's balance record?
+		acctdata, err := cow.lookup(txn.Txn.Sender)
+		if err != nil {
+			return err
+		}
+		correctAuthorizer := acctdata.SpendingKey
+		if (correctAuthorizer == basics.Address{}) {
+			correctAuthorizer = txn.Txn.Sender
+		}
```

Comment:
I'm not sure if this is a good idea, but consider adding something like `Authorizer()` to not just `SignedTxn` but also `AccountData`.